diff --git a/declarativeimports/core/environment.cpp b/declarativeimports/core/environment.cpp
index 2312e1d..f485893 100644
--- a/declarativeimports/core/environment.cpp
+++ b/declarativeimports/core/environment.cpp
@@ -60,13 +60,8 @@ uint Environment::makeVersion(uint major, uint minor, uint release) const
 
 uint Environment::identifyPlasmaDesktopVersion()
 {
-    //! Identify Plasma Desktop version
-    QProcess process;
-    process.start("plasmashell", QStringList() << "-v");
-    process.waitForFinished();
-    QString output(process.readAllStandardOutput());
-
-    QStringList stringSplit = output.split(" ");
+    qDebug() << "Plasma version:" << "@plasmaVersion@";
+    QStringList stringSplit = {"", "@plasmaVersion@"};
 
     if (stringSplit.count() >= 2) {
         qDebug() << " /////////////////////////";
