diff --git a/lua/cmp_digraphs.lua b/lua/cmp_digraphs.lua
index 0b71935..1ce91d5 100644
--- a/lua/cmp_digraphs.lua
+++ b/lua/cmp_digraphs.lua
@@ -12,7 +12,7 @@ function source.new()
 end

 function source:get_keyword_pattern()
-  return [[.\{1,2}]]
+  return [[-.\{1,2}]]
 end

 function source:complete(params, callback)
@@ -50,7 +50,7 @@ function source:_get_completion_items(opts)
       items[#items + 1] = {
         label = digraph .. ' ' .. vim.fn.strtrans(char),
         labelDetails = { detail = string.format('U+%04X', charnr) },
-        filterText = digraph,
+        filterText = '-' .. digraph,
         insertText = char,
       }
     end
