diff --git a/_nixos-rebuild b/_nixos-rebuild
index 84e8d22..d918e68 100644
--- a/_nixos-rebuild
+++ b/_nixos-rebuild
@@ -2,7 +2,7 @@
 
 # We're faking a `nix build` command-line to re-use Nix's own completion
 # for the few options passed through to Nix.
-_nixos-rebuild_pretend-nix() {
+_nom-rebuild_pretend-nix() {
   COMP_LINE="nix build ${COMP_LINE}"
   # number of prepended chars
   (( COMP_POINT = COMP_POINT + 10))
@@ -16,7 +16,7 @@ _nixos-rebuild_pretend-nix() {
   _complete_nix "nix"
 }
 
-_nixos-rebuild() {
+_nom-rebuild() {
   local curr="$2"
   local prev="$3"
   local subcommandGiven=0
@@ -128,28 +128,28 @@ _nixos-rebuild() {
       ;;
 
     -I)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
     --builders)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
     --flake)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
     --override-input)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
     --override-input_2)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
     --log-format)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
     --option)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
     --option_2)
-      _nixos-rebuild_pretend-nix
+      _nom-rebuild_pretend-nix
       ;;
 
     *)
@@ -162,4 +162,4 @@ _nixos-rebuild() {
   esac
 }
 
-complete -F _nixos-rebuild nixos-rebuild
+complete -F _nom-rebuild nom-rebuild
