diff --git a/lua/nvim-lightbulb/init.lua b/lua/nvim-lightbulb/init.lua
index 20c1ab9..1fc1d15 100644
--- a/lua/nvim-lightbulb/init.lua
+++ b/lua/nvim-lightbulb/init.lua
@@ -10,7 +10,7 @@ local LIGHTBULB_VIRTUAL_TEXT_NS = vim.api.nvim_create_namespace("nvim-lightbulb"
 
 -- Set default sign
 if vim.tbl_isempty(vim.fn.sign_getdefined(SIGN_NAME)) then
-  vim.fn.sign_define(SIGN_NAME, { text = "💡", texthl = "LspDiagnosticsDefaultInformation" })
+  vim.fn.sign_define(SIGN_NAME, { text = "", texthl = "LspDiagnosticsDefaultInformation" })
 end
 
 --- Update lightbulb float.
